### PR TITLE
Fix badge wrap

### PR DESCRIPTION
**Pull request by confused-Techie**
Currently the wrapping of badges can be a little awkward depending on screen size.

So this PR adds wrapping to avoid these issues.

## Before PR 
image

## After PR
image

### CODE REVIEW

**Comment by Daeraxa** [comment]
Sorry I totally forgot to review this one. I think the fix is fine for the moment but one issue is that you can see in the second image that it is clipping into the `p`. I'm wondering if we redo the package cards so that badges are always displayed on a second line and never inline with the title? Not sure if we would also want to extend that to the package view?

**Comment by confused-Techie** [comment]
A new span sounds right, but I'd double check to be sure.
But while we totally could just make a new line for the badges, for super short titles, I still feel like it's nice to be on the same line.

At the very least we will have to increase the top margin, to ensure it doesn't click like you pointed out.

And my bad for merging before review, just thought there was no way I could miss something on a `+1|-1` but looks like I did lol.

Appreciate the feedback, we will have to take a look at this. Also a sidenote, this same behavior is likely also reproducible within the `settings-view` package

**Comment by Daeraxa** [comment]
I'm still somewhat torn about the placement and I don't know the answer. I know exactly what you mean and I think keeping it on one line still looks better for the main package view. However the package cards end up looking a bit disjointed with the inconsistent placement:

image

Maybe instead of placing the badge next to the title they could be fixed to the right?

image

Although this looks even worse for badges that have been displaced to the second line:
image

On the other hand thinking back to putting it on its own line like I mentioned before, that ends up also being weird and stretches that row. (Actually this this is a little odd in its own right, I assumed the cards had a fixed height):

image

In summary, yeah, not sure lol. Need to think more on this.

**Comment by confused-Techie** [comment]
Tbh I thought they were a fixed height to lol.
Well, there's a few things we could do. And we might be looking at a bit of a bigger change as a solution.
Essentially increasing the top margin, and maybe attempting to find a way to force the cards to only show the badge on the line below, and have a max of how many lines they can occupy, then we additionally have the full view show them on the same line.

Something like that could work, although I'm pretty sure the same EJS that builds the preview cards, also builds the full detail, so might be easier said than done to make that portion of the change

**Comment by Daeraxa** [comment]
In the short term there are two things that need to be fixed:
1) The top margin
2) The random bit of leftover border on the first line which is an even worse problem than I thought. Why?

image

That is Firefox on the left and Chrome on the right. Go figure...

**Comment by confused-Techie** [comment]
Well, I didn't realize you were getting the little snippet of badge colour still. Thought that was resolved after my change.

So if you don't mind me, I'm gonna convert this all to pure HTML styling to avoid having to worry about CSS browser compatibility.

But on a serious note, yeah good point those are the biggest things to worry about right now, beyond that additional improvements should still be made, but can be considered separate enough

**Comment by Daeraxa** [comment]
Not sure if it is related to this topic but the featured page cards don't show badges at all?

image

**Comment by confused-Techie** [comment]
Not related to this concern, but yeah they don't for some reason.

The cached data of the featured packages isn't showing anything. But they really should? The cache was last updated yesterday, which the should contain any badge data there. Lemme check a few things on that.

But in other news, @Daeraxa try using 

```css
white-space: nowrap;
display: inline-block;
```

On the `.badge` class CSS, that seems to fix issues in both Chromium and Firefox for me, but doesn't yet address the spacing issue, but does stop them from at least overlapping, now they just ever so slightly touch

**Comment by Daeraxa** [comment]
Yeah that is looking a lot better:

image

**Comment by confused-Techie** [comment]
@Daeraxa perfect, I'll do a PR for that!
